cpp/Ice/cancellation/Client.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "Greeter.h"
+
+#include <cstdlib>
+#include <chrono>
+#include <iostream>
+
+using namespace std;
+
+int
+main(int argc, char* argv[])
+{
+    // Figure out my name.
+    const char* name = getenv("USER");
+    if (name == nullptr)
+    {
+        name = getenv("USERNAME");
+    }
+    if (name == nullptr)
+    {
+        name = "masked user";
+    }
+
+    // Set the Ice.ImplicitContext property to "Shared" before creating the communicator.
+    // This is only necessary for the implicit context API (see below).
+    Ice::InitializationData initData{.properties = Ice::createProperties(argc, argv)};
+    initData.properties->setProperty("Ice.ImplicitContext", "Shared");
```

Comment:
Do we need this in the cancellation demo?

Reply:
No, copy and paste leftover.